### ISSUE TITLE
client error [object Object] in AdminGuesser (created by AdminGuesserWithError) in API Platform Admin

### ISSUE THREAD

**Issue by McDuck** [api-platform/api-platform#1521]
**API Platform version(s) affected**: 2.5.5

**Description**  
I encounter an error in the API Platform Admin when I try out API Platform using the [Getting Started](https://api-platform.com/docs/distribution/) documentation.

**How to reproduce**
```
user@<host>:~$ uname -a
Linux <host> 5.0.0-1036-azure #38-Ubuntu SMP Sun Mar 22 21:27:21 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
user@<host>:~$ docker --version
Docker version 19.03.8, build afacb8b7f0
user@<host>:~$ docker-compose --version
docker-compose version 1.17.1, build unknown
user@<host>:~$ git clone git clone https://github.com/api-platform/api-platform.git
user@<host>:~$ cd api-platform
user@<host>:~$ git checkout v2.5.5
HEAD is now at ef0659e Bump dependencies

<add domain to trusted hosts>
user@<host>:~/api-platform$ git diff
diff --git a/api/config/packages/framework.yaml b/api/config/packages/framework.yaml
index 6089f4b..72fd36c 100644
--- a/api/config/packages/framework.yaml
+++ b/api/config/packages/framework.yaml
@@ -1,5 +1,9 @@
 framework:
     secret: '%env(APP_SECRET)%'
+    trusted_hosts:
+            - localhost
+            - <ipv4 address>
+            - <domain>
     #csrf_protection: true
     #http_method_override: true

<add .env because of slow VM>
user@<host>:~/api-platform$ cat .env
DOCKER_CLIENT_TIMEOUT=600
COMPOSE_HTTP_TIMEOUT=600

<I already pulled them, but the output tells something atleast>
user@<host>:~/api-platform$ sudo docker-compose pull
Pulling admin (quay.io/api-platform/admin:latest)...
latest: Pulling from api-platform/admin
Digest: sha256:eeb1bd42aec0434fa7cf7b961bf1362e008771b823f2643fc2b797959bb2d9a9
Status: Image is up to date for quay.io/api-platform/admin:latest
Pulling db (postgres:12-alpine)...
12-alpine: Pulling from library/postgres
Digest: sha256:9ea72265275674225b1eaa2ae897dd244028af4ee7ef6e4e89fe474938e0992e
Status: Image is up to date for postgres:12-alpine
Pulling client (quay.io/api-platform/client:latest)...
latest: Pulling from api-platform/client
Digest: sha256:6ce8e86a9cfcd0833389d8c82eb592e535d798fb048685502cca93d45c1767e9
Status: Image is up to date for quay.io/api-platform/client:latest
Pulling php (quay.io/api-platform/php:latest)...
latest: Pulling from api-platform/php
Digest: sha256:f373fa3d31f5917502ce36d52ee92fa88603efdb61247bf26041aa9ed43ff1f4
Status: Image is up to date for quay.io/api-platform/php:latest
Pulling api (quay.io/api-platform/nginx:latest)...
latest: Pulling from api-platform/nginx
Digest: sha256:e994c02ae115f18963e1fad3cfd6d3c9c3442874a7960f985ca6398b38818918
Status: Image is up to date for quay.io/api-platform/nginx:latest
Pulling mercure (dunglas/mercure:latest)...
latest: Pulling from dunglas/mercure
Digest: sha256:a0360e8317946ee49b1a5f8400e80b6ce2a9932880a9c1d3eebb070e49eeccb7
Status: Image is up to date for dunglas/mercure:latest
Pulling vulcain (dunglas/vulcain:latest)...
latest: Pulling from dunglas/vulcain
Digest: sha256:127abd739bd2322fbfb06646d2ec3ded85ce4afc10debe352661781605b91a4d
Status: Image is up to date for dunglas/vulcain:latest
user@<host>:~/api-platform$ sudo docker-compose up -d
Creating network "apiplatform_default" with the default driver
Creating apiplatform_db_1 ...
Creating apiplatform_dev-tls_1 ...
Creating apiplatform_db_1
Creating apiplatform_dev-tls_1 ... done                                                                 Creating apiplatform_mercure_1 ...
Creating apiplatform_client_1 ...
Creating apiplatform_admin_1 ...
Creating apiplatform_php_1 ...
Creating apiplatform_admin_1
Creating apiplatform_client_1
Creating apiplatform_mercure_1
Creating apiplatform_php_1 ... done                                                                     Creating apiplatform_api_1 ...
Creating apiplatform_api_1 ... done                                                                     Creating apiplatform_vulcain_1 ...
Creating apiplatform_admin_1 ... done
```

From a remote laptop using Chrome visit: https://\<domain\>:8443/
First shows the loading screen.
```
Loading
The page is loading, just a moment please.
```
Then shows the following.
```
Something went wrong
A client error occurred and your request couldn't be completed.
[object Object]

    in AdminGuesser (created by AdminGuesserWithError)
    in ErrorBoundary (created by AdminGuesserWithError)
    in TranslationProvider (created by AdminGuesserWithError)
    in AdminGuesserWithError (created by HydraAdmin)
    in HydraAdmin (at App.js:4)
    in Unknown (at src/index.js:6)
```
Error in Chrome DevTools Network:
```
<domain> | 200 | document | Other | 1.2 kB | 233 ms
-- | -- | -- | -- | -- | --
bundle.js | 200 | script | (index) | 6.7 kB | 54 ms
0.chunk.js | 200 | script | (index) | 1.3 MB | 1.28 s
main.chunk.js | 200 | script | (index) | 2.8 kB | 423 ms
sockjs-node | 101 | websocket | webpackHotDevClient.js:60 | 0 B | Pending
localhost | (failed)net::ERR_CONNECTION_REFUSED | fetch | VM88:1 | 0 B | 4.01 s
favicon.ico | 200 | x-icon | Other | 7.0 kB | 86 ms
0.chunk.js | 200 | fetch | VM88:1 | 1.3 MB | 637 ms
0.chunk.js.map | 200 | fetch | VM88:1 | 1.3 MB | 661 ms
```
While `localhost` tooltips as `https://localhost:8443/` while others tooltip as `<protocol>://<domain>:444/<path>`

**Possible Solution**  
Don't use `localhost` in client admin, but use `document.domain`, because `https://<domain>:8443/` is where the API of API Platform is located.

**Additional Context**  
Loading screen:
![image](https://user-images.githubusercontent.com/1067670/80861180-25b27e00-8c6d-11ea-8c89-4693d22c434b.png)
Client error:
![image](https://user-images.githubusercontent.com/1067670/80861120-c2285080-8c6c-11ea-836b-9b9fa5dc4deb.png)
Chrome DevTools console:
![image](https://user-images.githubusercontent.com/1067670/80861237-79bd6280-8c6d-11ea-99a3-66b3a96f36c4.png)
Chrome DevTools Network:
![image](https://user-images.githubusercontent.com/1067670/80861288-f05a6000-8c6d-11ea-8e44-a1a70e50c609.png)

Chrome DevTools Console:
```
DevTools failed to load SourceMap: Could not load content for chrome-extension://hdokiejnpimakedhajhdlcegeplioahd/sourcemaps/onloadwff.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
log.js:24 [HMR] Waiting for update signal from WDS...
react-dom.development.js:24994 Download the React DevTools for a better development experience: https://fb.me/react-devtools
react.development.js:315 Warning: React.createFactory() is deprecated and will be removed in a future major release. Consider using JSX or use React.createElement() directly instead.
printWarning @ react.development.js:315
warn @ react.development.js:278
createFactoryWithValidation @ react.development.js:1827
(anonymous) @ shouldUpdate.js:25
pure @ pure.js:29
./node_modules/ra-core/esm/util/FieldTitle.js @ FieldTitle.js:14
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/util/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/sideEffect/useUnselectAll.js @ useUnselectAll.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/sideEffect/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/auth/useLogoutIfAccessDenied.js @ useLogoutIfAccessDenied.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/auth/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/core/CoreAdminContext.js @ CoreAdminContext.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/core/CoreAdmin.js @ CoreAdmin.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/core/components.js @ components.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/core/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/actions/dataActions/crudCreate.js @ crudCreate.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/actions/dataActions/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/actions/accumulateActions.js @ accumulateActions.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/actions/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/reducer/admin/resource/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/reducer/admin/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/reducer/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/ra-core/esm/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/react-admin/esm/AdminContext.js @ AdminContext.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/react-admin/esm/Admin.js @ Admin.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/react-admin/esm/index.js @ index.js:1
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/@api-platform/admin/lib/AdminGuesser.js @ AdminGuesser.js:12
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./node_modules/@api-platform/admin/lib/index.js @ index.js:91
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./src/App.js @ main.chunk.js:14
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
./src/index.js @ App.js:4
__webpack_require__ @ bootstrap:784
fn @ bootstrap:150
1 @ serviceWorker.js:141
__webpack_require__ @ bootstrap:784
checkDeferredModules @ bootstrap:45
webpackJsonpCallback @ bootstrap:32
(anonymous) @ main.chunk.js:1
Show 55 more frames
VM77:1 GET https://localhost:8443/ net::ERR_CONNECTION_REFUSED
(anonymous) @ VM77:1
(anonymous) @ fetchJsonLd.ts:14
step @ tslib.es6.js:100
(anonymous) @ tslib.es6.js:81
(anonymous) @ tslib.es6.js:74
__awaiter @ tslib.es6.js:70
fetchJsonLd @ fetchJsonLd.ts:9
(anonymous) @ parseHydraDocumentation.ts:77
step @ tslib.es6.js:100
(anonymous) @ tslib.es6.js:81
(anonymous) @ tslib.es6.js:74
__awaiter @ tslib.es6.js:70
fetchEntrypointAndDocs @ parseHydraDocumentation.ts:70
parseHydraDocumentation @ parseHydraDocumentation.ts:179
introspect @ dataProvider.js:423
(anonymous) @ AdminGuesser.js:136
commitHookEffectListMount @ react-dom.development.js:19731
commitPassiveHookEffects @ react-dom.development.js:19769
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
flushPassiveEffectsImpl @ react-dom.development.js:22853
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushPassiveEffects @ react-dom.development.js:22820
(anonymous) @ react-dom.development.js:22699
workLoop @ scheduler.development.js:597
flushWork @ scheduler.development.js:552
performWorkUntilDeadline @ scheduler.development.js:164
AdminGuesser.js:146 Uncaught {api: Api, error: TypeError: Failed to fetch, response: undefined, status: undefined}
(anonymous) @ AdminGuesser.js:146
basicStateReducer @ react-dom.development.js:15013
updateReducer @ react-dom.development.js:15135
updateState @ react-dom.development.js:15237
useState @ react-dom.development.js:15949
useState @ react.development.js:1497
AdminGuesser @ AdminGuesser.js:116
renderWithHooks @ react-dom.development.js:14803
updateFunctionComponent @ react-dom.development.js:17034
beginWork @ react-dom.development.js:18610
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
beginWork$1 @ react-dom.development.js:23203
performUnitOfWork @ react-dom.development.js:22157
workLoopSync @ react-dom.development.js:22130
performSyncWorkOnRoot @ react-dom.development.js:21756
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
scheduleUpdateOnFiber @ react-dom.development.js:21199
dispatchAction @ react-dom.development.js:15660
(anonymous) @ AdminGuesser.js:145
Promise.catch (async)
(anonymous) @ AdminGuesser.js:143
commitHookEffectListMount @ react-dom.development.js:19731
commitPassiveHookEffects @ react-dom.development.js:19769
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
flushPassiveEffectsImpl @ react-dom.development.js:22853
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushPassiveEffects @ react-dom.development.js:22820
(anonymous) @ react-dom.development.js:22699
workLoop @ scheduler.development.js:597
flushWork @ scheduler.development.js:552
performWorkUntilDeadline @ scheduler.development.js:164
index.js:1 The above error occurred in the <AdminGuesser> component:
    in AdminGuesser (created by AdminGuesserWithError)
    in ErrorBoundary (created by AdminGuesserWithError)
    in TranslationProvider (created by AdminGuesserWithError)
    in AdminGuesserWithError (created by HydraAdmin)
    in HydraAdmin (at App.js:4)
    in Unknown (at src/index.js:6)

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary.
console.<computed> @ index.js:1
logCapturedError @ react-dom.development.js:19527
logError @ react-dom.development.js:19564
callback @ react-dom.development.js:20744
callCallback @ react-dom.development.js:12490
commitUpdateQueue @ react-dom.development.js:12511
commitLifeCycles @ react-dom.development.js:19858
commitLayoutEffects @ react-dom.development.js:22803
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
commitRootImpl @ react-dom.development.js:22541
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
commitRoot @ react-dom.development.js:22381
finishSyncRender @ react-dom.development.js:21807
performSyncWorkOnRoot @ react-dom.development.js:21793
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
scheduleUpdateOnFiber @ react-dom.development.js:21199
dispatchAction @ react-dom.development.js:15660
(anonymous) @ AdminGuesser.js:145
Promise.catch (async)
(anonymous) @ AdminGuesser.js:143
commitHookEffectListMount @ react-dom.development.js:19731
commitPassiveHookEffects @ react-dom.development.js:19769
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
flushPassiveEffectsImpl @ react-dom.development.js:22853
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushPassiveEffects @ react-dom.development.js:22820
(anonymous) @ react-dom.development.js:22699
workLoop @ scheduler.development.js:597
flushWork @ scheduler.development.js:552
performWorkUntilDeadline @ scheduler.development.js:164
index.js:1 Warning: Missing translation for key: "[object Object]"
```

**Comment by McDuck** [comment]
And ofcourse when you deepdive into the error... I just found a workaround which is good enough for developing:
```
sudo -E ssh -F ~/.ssh/config -L444:localhost:444 -L8443:localhost:8443 <host>
```
From the remote laptop using Chrome visiting https://localhost:8443/ works! :-)

But ofcourse I think this should be fixed for the ease of getting started for people like me.

**Comment by Faber-smythe** [comment]
Hi, trying to figure things out on my end. I'm having exactly the same error and output as you showed. 

I don't understand how you fixed it though. 
I'd really appreciate some explanation about why it can't reach the API, and how to sort it out. I'm on windows though, so I'm not sure how to translate your commandline.

**Comment by McDuck** [comment]
I think the reason is that https://github.com/api-platform/api-platform/blob/v2.5.5/admin/.env contains the hardcoded string `https://localhost:8443` and not a dynamic hostname.

You could try if you change this value if this works for you. I tried it shortly, but I didn't succeed. I didn't try hard though. ;-)

I solved it for myself creating [local port forwarding using a SSH tunnel](https://help.ubuntu.com/community/SSH/OpenSSH/PortForwarding#Local_Port_Forwarding) on my Windows 10 laptop using [WSL](https://docs.microsoft.com/nl-nl/windows/wsl/install-win10) with the [Ubuntu app](https://www.microsoft.com/en-us/p/ubuntu/9nblggh4msv6).
Probably you can make this work using [Putty](https://www.akadia.com/services/ssh_putty.html) as well.

**Comment by Faber-smythe** [comment]
Thanks for the reply, I'll look into it !
I'll post a feedback once I get some result. =)

**Comment by Faber-smythe** [comment]
Okay, so in the end, it seems that it was only a matter of **cross-origin headers**.

Maybe we didn't have the same issue afterall, but the output looked exactly the same for me :

![image](https://user-images.githubusercontent.com/39943985/82060443-b2c2f180-96c7-11ea-918f-75d7383cf73e.png)

**So to solve this**, I used [nelmio/cors-bundle](https://github.com/nelmio/NelmioCorsBundle). It added a `CORS_ALLOW_ORIGIN` in the .env file.
I just had to adjust to my localhost pattern - later to be edited to match the target API's domain. 
That did the trick for me.

Anyway, thanks for your time and your answer.  :)

**Comment by bhtomming** [comment]
I have the same error. How did you solve it?
![QQ截图20200524145828](https://user-images.githubusercontent.com/24985179/82747799-623a4b00-9dcf-11ea-9229-7f3407aef53b.png)

**Comment by Faber-smythe** [comment]
I'll try to be more specific.
I first did a lot of research, since the debug output was of no help at all. At some point, seeing nothing workd I just posted here to see if McDuck's case would help mine. 

And while I waited for a reply from him, I just tried to redo a minimalistic project (single endpoint API on port 8000, and admin tool on port 3000), see if the error reproduced.
It did, until I **fixed the CORS headers**. 
See, the API will decline requests from remote origins (_ie_ not from local port 8000, in my case), unless you explicitely accept foreign domains.

There is a bundle that helps you do just that for Symfony, the one I quoted just above, in my previous answer. So in your Symfony API, you'll need to require [nelmio/cors-bundle](https://github.com/nelmio/NelmioCorsBundle). Then adjust this line that should have been added at the end of your .env file.
In order to match any localhost domain, the pattern goes like this :
`CORS_ALLOW_ORIGIN=^https?://(localhost|127\.0\.0\.1)(:[0-9]+)?$`

So that solved the issue for my new fake project. With the wrong pattern I got this error (screenshot above), with the right one (or just `*` to match any domain) I didn't.
**This fix didn't work for my actual project though**. I tried to deconstruct it little by little to see where lied the thing that caused it, but I didn't find it. 

So in the end, I just took the file from my project I knew I needed, and put them in my "fake" API.
That did the trick for me but I'm nothing close to an expert on api/platform or even Symfony. 
Still I hope it helps out. :)
